Assets/Scripts/UI/LevelSelect.cs

```diff
@@ -86,15 +102,63 @@ private void LazyLoad()
             FindObjectOfType<SaveDataManager>()?.PreloadSaveData();
         }
 
-        private void OnLevelSelected(string sceneName, Button clickedButton)
+        // <summary>
+        // retrieves the name of the level from the cutscene name
+        // sceneName must be one of: "IntroCutscene" "SecondCutscene" "ThirdCutscene" "FourthCutscene"
+        // <summary>
+        private string GetLevelName(string sceneName)
+        {
+            if (!allLevelLoadingSceneNames.Contains(sceneName))
+            {
+                return "";
+            }
+            int idx = allLevelLoadingSceneNames.IndexOf(sceneName);
+            return allLevelSceneNames[idx];
+        }
+
+        private string GetCorrectText(string sceneName, string type, int levelNumber)
```

Comment:
frankly since all the types' logic is fairly distinct and you use it only once below you might as well just not have this method and inline the logic as it'll be significantly simpler